pkg/reconciler/dependencybuild/dependencybuild.go

```diff
@@ -309,6 +313,262 @@ func (r *ReconcileDependencyBuild) handleStateBuilding(ctx context.Context, db *
 	tr.Labels = map[string]string{artifactbuild.DependencyBuildIdLabel: db.Labels[artifactbuild.DependencyBuildIdLabel], artifactbuild.TaskRunLabel: "", TaskType: TaskTypeBuild}
 
 	tr.Spec.TaskRef = &pipelinev1beta1.TaskRef{Name: db.Status.CurrentBuildRecipe.Task, Kind: pipelinev1beta1.ClusterTaskKind}
+	switch {
+	case db.Status.CurrentBuildRecipe.Maven:
+		tr.Spec.TaskSpec = &pipelinev1beta1.TaskSpec{
```

Comment:
Idea for discussion: would not be easier to have it as yaml file which will be [embed](https://pkg.go.dev/embed)?
Pros:
-  Easy to read/edit
- Easy to test standalone

Cons:
- No validation at build time

Reply:
ok the solution I'm about to push avoids creating yaml files that we do not want included in the infra-deployment etc. processing set, but uses yaml to construct these huge tasks.

Let's see what you all think.